src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -100,3 +107,105 @@ export function useUSDCValue(currencyAmount?: CurrencyAmount<Currency>) {
     }
   }, [currencyAmount, error, price])
 }
+
+/**
+ * Returns the price in USDC of the input currency from price APIs
+ * @param currencyAmount currency to compute the USDC price of
+ */
+export function useUSDCValue(currencyAmount?: CurrencyAmount<Currency>) {
+  const usdcPrice = useUSDCPrice(currencyAmount?.currency)
+
+  return useGetPriceQuote({ ...usdcPrice, currencyAmount })
+}
+
+export function useCoingeckoUsdPrice({ tokenAddress }: Pick<CoinGeckoUsdPriceParams, 'tokenAddress'>) {
+  // default to MAINNET (if disconnected e.g)
+  const { chainId = DEFAULT_NETWORK_FOR_LISTS } = useActiveWeb3React()
+  const [price, setPrice] = useState<Price<Token, Currency> | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+
+  const currency = useToken(tokenAddress)
+
+  useEffect(() => {
+    getUSDPriceQuote({
+      chainId,
+      tokenAddress,
+    })
+      .then(toPriceInformation)
+      .then((response) => {
```

Comment:
just curious, why do you expose the original data instead of just the, priceInformation directly?

Reply:
This `response` i think is to generic name. maybe `priceInfo`?

Reply:
i think it was because in the other APIs it's done similarly ( i think ) - and for paraswap there are other useful props that could posisbly want to be used elsewhere before the `priceInformation` schema mapping. In the case of coingecko there isn't any useful info so i could change the fn directly 👍